### PR TITLE
Avoid explicit test for package.json exists

### PR DESCRIPTION
**Pull request by shayne-fletcher**
Implement the advice of https://github.com/digital-asset/daml/pull/5001#discussion_r392466634: catch the error from reading a package.json that doesn't exist in that case, proceed with a fresh `PackageJson` value.

### CODE REVIEW

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 1]
```diff
@@ -730,32 +731,27 @@ instance ToJSON PackageJson where
...
         -- Topologically order our packages.
       outBaseDir = T.pack $ takeFileName optOutputDir
         -- The leaf directory of the output directory (e.g. often 'daml2ts').
   let ourWorkspaces = map ((outBaseDir <> "/") <>) ps
-  bytes <- BSL.readFile optInputPackageJson
-  case decode bytes :: Maybe PackageJson of
-    Nothing -> fail $ "Error decoding JSON from '" <> optInputPackageJson <> "'"
-    Just oldPackageJson -> transformAndWrite ourWorkspaces outBaseDir oldPackageJson
+  mbBytes <- catch (Just <$> BSL.readFile optInputPackageJson)
+             (const (pure Nothing) :: IOException -> IO (Maybe BSL.ByteString))
```
Comment (cocreature) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]:
```suggestion
```suggestion
  mbBytes <- catchJust (guard . isDoesNotExistError) (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
```

Comment (shayne-fletcher) [reply language-support/ts/codegen/src/TsCodeGenMain.hs]: Nice.

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 2]
```diff
@@ -730,32 +737,27 @@ instance ToJSON PackageJson where
...
   let ourWorkspaces = map ((outBaseDir <> "/") <>) ps
-  bytes <- BSL.readFile optInputPackageJson
-  case decode bytes :: Maybe PackageJson of
-    Nothing -> fail $ "Error decoding JSON from '" <> optInputPackageJson <> "'"
-    Just oldPackageJson -> transformAndWrite ourWorkspaces outBaseDir oldPackageJson
+  mbBytes <- catchJust (guard . isDoesNotExistError)
+    (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
+  packageJson <- case mbBytes of
+    Nothing -> pure mempty
+    Just bytes -> case decode bytes :: Maybe PackageJson of
```
Comment (hurryabit) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: I think the following is easier to read (and shorter):
```suggestion
    Just bytes -> case decode @PackageJson bytes of
```

File: language-support/ts/codegen/src/TsCodeGenMain.hs [hunk 3]
```diff
@@ -730,32 +737,27 @@ instance ToJSON PackageJson where
...
         -- The leaf directory of the output directory (e.g. often 'daml2ts').
   let ourWorkspaces = map ((outBaseDir <> "/") <>) ps
-  bytes <- BSL.readFile optInputPackageJson
-  case decode bytes :: Maybe PackageJson of
-    Nothing -> fail $ "Error decoding JSON from '" <> optInputPackageJson <> "'"
-    Just oldPackageJson -> transformAndWrite ourWorkspaces outBaseDir oldPackageJson
+  mbBytes <- catchJust (guard . isDoesNotExistError)
+    (Just <$> BSL.readFile optInputPackageJson) (const $ pure Nothing)
+  packageJson <- case mbBytes of
+    Nothing -> pure mempty
```
Comment (hurryabit) [inline language-support/ts/codegen/src/TsCodeGenMain.hs]: I'm not a big fan of using (highly) polymorphic constants in monomorphic code. What is `mempty` here? Is it `[]` or `Sum 0` or `Endo id`? I _guess_ it's an empty string of some sort. If so, I'd prefer to have the following for the sake of legibility:
```suggestion
    Nothing -> pure ""
```

**Pull request merged**